core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -83,4 +85,33 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, Set<Long> validIds, Set<Long> expiredIds) {
```

Comment:
Rather than guessing which stats files were removed during expiration, I think that this should just diff the two lists of stats files.

Reply:
ok. I will update it.

Reply:
> just diff the two lists of stats files.

Two StatisticsFiles objects can use the same stats file location in case of reuse.  (for example during rewrite data files). I have simplified the logic now.